### DISCUSSION TITLE
qrcode-with-logos/guide/discussions

### DISCUSSION THREAD

**Discussion by giscus** [zxpsuper/qrcode-with-logos#35] [Announcements]
# qrcode-with-logos/guide/discussions

The tool for creating a QRcode with logo

https://zxpsuper.github.io/qrcode-with-logos/guide/discussions.html

**Comment by zxpsuper** [comment]
Welcome to qrcode-with-logos Discussions!😁

**Comment by jatin263** [comment]
i am not able to download the qr code

**Reply by jatin263** [reply]
using nodejs

**Reply by zxpsuper** [reply]
This library only work in the browser because it use the canvas to draw different style.

**Reply by zxpsuper** [reply]
Now you can use and download images in Node.js.

**Comment by zxpsuper** [comment]
Q: Loaded QRCode not sharp enough?
A: Maybe default width 380 is too small so that qrcode is not sharp enough. I set the width to 1000 and it looks better.

**Comment by zxpsuper** [comment]
If you found the logo is too small and can not adjust it, you can set `nodeQrCodeOptions.errorCorrectionLevel = "H"` and it will be larger than before.

**Comment by riskinurohman** [comment]
Hi. Your library is awesome. However i'm facing an issue with CORS in src. Could you help me please?

**Reply by zxpsuper** [reply]
You can set `logo.crossOrigin` to "anonymous" or "use-credentials" or "no-cors" according to your situation.

The details still depend on your target server's configuration of cors .

**Comment by fkrichene** [comment]
I want to bring up an issue I encountered regarding Content Security Policy (CSP) restrictions when the library sets the img.src property.

Currently, when the library sets the src attribute of an <img> element to a base64 data URL, it triggers a CSP violation on my site, as my CSP does not allow data: URIs. While this doesn't affect Canvas rendering directly, the setting of img.src causes a CSP error.

Would it be possible to make this behavior optional? For use cases like mine where rendering the QR code directly onto a Canvas is sufficient, avoiding the creation of the img element would help bypass CSP restrictions.

**Comment by net2use** [comment]
Well, a great library!
I small question to the logo path. I try to referred to an svg-Logo.
Maybe your lib is not prepared to read a svg-imagetype?

Thank you!
